msal/src/main/java/com/microsoft/identity/client/IPublicClientApplication.java

```diff
@@ -109,6 +110,58 @@ void acquireToken(@NonNull final Activity activity,
      */
     boolean isSharedDevice();
 
+    /**
+     * Signs the provided {@link PoPAuthenticationScheme} parameters into a JWT on behalf of the
+     * provided {@link IAccount}.
+     * <p>
+     * Important: Use of this API requires setting the minimum_required_broker_protocol_version to
+     * "6.0" or higher.
+     *
+     * @param account       The account for whom signing shall occur.
+     * @param popParameters The input parameters.
+     * @return The resulting SHR.
+     */
+    String generateSignedHttpRequest(@NonNull final IAccount account,
```

Comment:
This should never return null?

Reply:
what about empty strings?

Reply:
Neither are allowed -- either the SHR or an Exception, if there was a problem

1fc28b4a9972e91c3722b5b512ff7d98e3a3dca7